zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -468,13 +480,50 @@ pub(crate) fn put_received_transparent_utxo<P: consensus::Parameters>(
             output.mined_height(),
             address,
             receiving_account,
+            true,
         )
     } else {
         // The UTXO was not for any of our transparent addresses.
         Err(SqliteClientError::AddressNotRecognized(*address))
     }
 }
 
+/// Returns the vector of [`TransactionDataRequest`]s that represents the information needed by the
+/// wallet backend in order to be able to present a complete view of wallet history and memo data.
+pub(crate) fn transaction_data_requests<P: consensus::Parameters>(
+    conn: &rusqlite::Connection,
+    params: &P,
+) -> Result<Vec<TransactionDataRequest>, SqliteClientError> {
+    // We cannot construct transaction data requests for the case where we cannot determine the
+    // height at which to begin, so we require that either the target height or mined height be
+    // set. In practice, we should not encounter entries in `transparent_spend_search_queue`
+    // because under ordinary circumstances, it is populated via a call from
```

Comment:
Is there something missing here qualifying which entries we should not expect to encounter? Or does this mean that `transparent_spend_search_queue` is ordinarily empty?

Reply:
Yes, `transparent_spend_search_queue` will be empty except in wallet recovery scenarios.

Reply:
I guess the comment doesn't really help me because I don't know what situations are "ordinary". The comment at https://github.com/zcash/librustzcash/blob/8752ad7e19a93b5b9dbc2976daa2f7a5b2ec5a29/zcash_client_sqlite/src/lib.rs#L1446-L1450 (in the only place where `queue_transparent_spend_detection` is called) is more helpful. I don't think it is restricted to wallet recovery; it can also happen if another wallet shares the spending key for the output.

Reply:
While I don't believe that wallets sharing a seed should be considered an ordinary circumstance, it can also occur with imported viewing capabilities, so I agree, I will update this comment.